mantidproject/mantid
# Add support for Figaro instrument in LoadILLReflectometry

This PR enables `LoadILLReflectometry` to properly load Figaro NeXus files by adding proper handling of instrument geometry to the loader and fixing the IDFs of Figaro and D17. Additionally, this PR renames the *BeamPosition* property to *DirectBeamPosition*, and adds a new *BeamCentre* property. *BeamCentre* is meant to be used in situations where the automatic peak finding algorithm in the loader fails.

The API breakage should not be an issue since the loader is currently only used by the Mantid developers at ILL.

**To test:**

There is a single Figaro file available in the `ILL/Figaro/` subdirectory in unit test data. Try to load it with different combinations of properties and see if something breaks.

Fixes #21213.

**Release Notes** 

All the changes are mentioned on the reflectometry page.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
